### PR TITLE
Gobierto Data / fix dataset source & license with urls

### LINKED ISSUE #3032: Data / New fields for datasets: Source and License
image

For the time being we can specify the License in the dataset text. 

In the future: 

* **Source**: We should allow to specify a source (name and URL) for a dataset. This can be a custom field. We should also think if we should detail the license associated to the sourced datasets. 

* **License**: A license will be a name and URL. Possible License for Getafe: https://opendefinition.org/od/2.1/es/

### LINKED ISSUE #3865: Data / If source link is not present, don't link the source
When you only specify a source, but not the source link, just show the source but don't link it (since now it's linking to nowhere)

image

image

### PR DESCRIPTION
**Pull request by stbnrivas**
Closes #3032 (after reopen) #3865

## :v: What does this PR do?
Gobierto-Data
- show license always as a link if selected
- show source of dataset  text && || url 

## :mag: How should this be manually tested?
1. from backoffice delete: 
- vocabulary for `license` if exist
- custom fields `data` [`source`, `source link`, `source`]

2.- create vocabulary and custom fields from:
```
bin/rails c
site = Site.find_by(domain: "**********")
GobiertoSeeds::GobiertoData::Recipe.run(site) if site.configuration.gobierto_data_enabled?
```
3.- make sure the insertions was done

4. create datasets with license alternatives:
- license yes
- license no
visit every alternative 

5. create a dataset with combination source text and source url:
visit every alternative
- source text no source url no
- source text no source url yes
- source text yes source url no
- source text yes source url yes

### After this PR

Screenshot from 2021-05-03 16-28-15

## :shipit: Does this PR changes any configuration file?

- [ ] new environment variable in `.env.example`?
- [ ] new entry in `config/application.yml`?
- [ ] new entry in `config/secrets.yml`?

(Changes in these files might need to update the role in Ansible)

## :book: Does this PR require updating the documentation?

- [ ] new site configuration variable?
- [ ] new site template?
- [ ] new module/submodule settings?
- [ ] significant changes in some feature?

### CODE REVIEW

File: app/javascript/gobierto_data/webapp/components/commons/InfoBlockText.vue [hunk 1]
```diff
@@ -8,15 +8,44 @@
...
+        :href="licenseUrl | translate"
+        class="gobierto-data-summary-header-container-text"
+      >
+        {{ licenseText | translate }}
+      </a>
+    </template>
+    <template v-else-if="hasSource">
+      <template v-if="hasSourceUrl">
+        <a
+          :href="sourceUrl"
```
Comment (ferblape) [inline app/javascript/gobierto_data/webapp/components/commons/InfoBlockText.vue]: @jorgeatgu could you add target blank?

File: app/javascript/gobierto_data/webapp/components/commons/InfoBlockText.vue [hunk 2]
```diff
@@ -8,15 +8,44 @@
     <span class="gobierto-data-summary-header-container-label">
       {{ label }}
     </span>
-    <span class="gobierto-data-summary-header-container-text">
-      {{ text }}
-    </span>
+    <template v-if="hasLicense">
+      <a
+        :href="licenseUrl | translate"
```
Comment (ferblape) [inline app/javascript/gobierto_data/webapp/components/commons/InfoBlockText.vue]: @jorgeatgu could you add target blank?

File: app/javascript/gobierto_data/webapp/components/sets/SummaryTab.vue [hunk 3]
```diff
@@ -275,11 +277,21 @@ export default {
       data_updated_at: this.dateUpdated,
       category: [{ name_translations: this.category } = {}] = [],
       frequency: [{ name_translations: this.frequency } = {}] = [],
-      "dataset-license": [{ name_translations: this.datasetLicense } = {}] = [],
+      // "dataset-license": [{ name_translations: this.datasetLicense, description_translations: this.datasetLicenseUrl } = {}] = [],
```
Comment (ferblape) [inline app/javascript/gobierto_data/webapp/components/sets/SummaryTab.vue]: comment, please remove

File: app/javascript/gobierto_data/webapp/components/commons/Info.vue [hunk 4]
```diff
@@ -155,6 +148,12 @@ export default {
     },
     checkStringLength() {
       return this.descriptionDataset.length > 250
+    },
+    hasSourceDataset() {
```
Comment (ferblape) [inline app/javascript/gobierto_data/webapp/components/commons/Info.vue]: the second part of the condition should check url attribute?

**Review by ferblape — CHANGES_REQUESTED** [review]

File: app/javascript/gobierto_data/webapp/components/commons/Info.vue (hunk 4 above)
Comment (stbnrivas) [reply app/javascript/gobierto_data/webapp/components/commons/Info.vue]: not here, we are only interested into have a not empty text ... the checks about condition url are into InfoBlockText.vue

    hasSourceUrl() {
      return this.source.url !== undefined && this.source.url !== ""
    }

File: app/javascript/gobierto_data/webapp/components/commons/InfoBlockText.vue (hunk 1 above)
Comment (stbnrivas) [reply app/javascript/gobierto_data/webapp/components/commons/InfoBlockText.vue]: done

File: app/javascript/gobierto_data/webapp/components/commons/InfoBlockText.vue (hunk 2 above)
Comment (stbnrivas) [reply app/javascript/gobierto_data/webapp/components/commons/InfoBlockText.vue]: done

File: app/javascript/gobierto_data/webapp/components/commons/Info.vue (hunk 4 above)
Comment (ferblape) [reply app/javascript/gobierto_data/webapp/components/commons/Info.vue]: ok!

**Review by ferblape — APPROVED** [review]
Now works well, thanks! And also I can see your refactor to make it more generic. So let's merge!

**Pull request merged**